### ISSUE TITLE
sessions invisible after kill-session from switcher

### ISSUE THREAD

**Issue by pandalanax** [omerxx/tmux-sessionx#72]
Deleting a session makes other sessions invisible.
Example:
I have four sessions running.
![image](https://github.com/omerxx/tmux-sessionx/assets/38329801/529d2c3c-7ea0-43ae-887c-87bc13faab8d)

Now deleting test2 session leads to this:
![image](https://github.com/omerxx/tmux-sessionx/assets/38329801/fc18d94f-1477-4e69-ac71-a46616392015)

Reopening the session view is listing all sessions again:
 
![image](https://github.com/omerxx/tmux-sessionx/assets/38329801/00b4c671-fa16-4a51-a53d-783623e2d77d)

Tmux Version: 3.3a
tmux-sessionx running on e6d0215 
nixOS 23.11

**Comment by omerxx** [comment]
Yep, what's happening here is you killing the current session "test2", tmux puts you on the next one on the list "test1" and sessionx reverting to its default of filtering out the current session from the list.

Note to self: `KILL_SESSION` invokes `reload_sessions.sh` where the code should take user configuration into consideration

**Comment by pandalanax** [comment]
mh, in the screenshots my current session is "nix". But if i am in "test2" and kill it, the behavior you described is correct.
I have ` set-option -g detach-on-destroy off` set in my tmux.conf

**Comment by sarojregmi200** [comment]
@omerxx, I got what you said. And was thinking about opening an issue regarding this problem.

```bash
set -g @sessionx-filter-current 'false'
```
Works as configured, and when any one of the session is deleted it removes the session but ignores the above configuration and hides the session that is currently active. 

> Thanks for the note above as it cleared me where to look for the problem.

```bash
if [[ $(echo "$SESSIONS" | wc -l) -gt 1 ]]; then
	echo "$SESSIONS" | grep -v "$CURRENT_SESSION"
else
	echo "$SESSIONS"
fi
```
Here the configuration check is missing, I would love to look at it. 
I think this issue has been open for long.

**Comment by sarojregmi200** [comment]
@omerxx Just a gentle reminder. What are your thoughts on this?
I think you missed the above mention.

**Comment by omerxx** [comment]
@sarojregmi200 Sorry for the delay, pushed a branch held in a PR (#98), this should solve the issue.
Would you like to quickly test it by installing the plugin from that branch?
```
set -g @plugin 'omerxx/tmux-sessionx#72-kill-and-filter'
```

**Comment by sarojregmi200** [comment]
I will once I am free. 

And would like to see your approach, I have already patched my cloned version.

**Comment by omerxx** [comment]
> I will once I am free.
> 
> And would like to see your approach, I have already patched my cloned version.

@sarojregmi200 It's just the same statement used on the main .sh file.
Since fzf's `execute` requires an executable to run, I had to separate `reload-sessions.sh` form the main plugin, creating a duplication. A more elegant approach would be another `utils` file that I could `source` on both cases but for now this should be enough

**Comment by sarojregmi200** [comment]
Yeah! I was also thinking the same looking at the changed files. And am yet to test it but I think there are some more configurations that should be duplicated to ensure user's customization takes effect.
Let me confirm my guess and report it here.
